### PR TITLE
fix(examples): ignore directories named .DS_Store in metro.config

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes

This change filters out directories named .DS_Store in the metro.config.js in our react-native example so that it doesn't try to treat it as a package.

When running `yarn react-native-example ios`, this error was encountered:

```
error Cannot find module '/Volumes/workplace/amplify-ui/packages/.DS_Store/package.json'
```
Screen Shot 2023-01-23 at 12 07 32 PM

#### Description of how you validated changes

After the change,  `yarn react-native-example ios` runs successfully.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/react-native/metro.config.js [hunk 1]
```diff
@@ -31,12 +31,14 @@ function getInternalPackages(root, internalPackagesDirectory) {
...
-        return { packageName, packagePath };
-      });
+      })
+        .filter(({ name }) => name !== '.DS_Store')
+        .map(({ name }) => {
+          const packagePath = path.resolve(internalDirectoryRoot, name);
+          const packageName = require(`${packagePath}/package.json`).name;
+
+          return { packageName, packagePath };
+        });
```
Comment (dbanksdesign) [inline examples/react-native/metro.config.js:41]: Could we make this logic a bit more robust to check if instead there exists a package.json at the packagePath? We could do a try/catch or filtering if that file exists using fs.access or something similar?

**Review by dbanksdesign — COMMENTED** [review]
Thanks for adding this, just one small comment to make this more robust for future use cases

File: examples/react-native/metro.config.js (hunk 1 above)
Comment (hbuchel) [reply examples/react-native/metro.config.js:41]: Yeah, good point, let me take a look!

Comment (zchenwei) [reply examples/react-native/metro.config.js:41]: +1

**Pull request closed without merging**